exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -1,10 +1,194 @@
+from typing import Tuple, Dict, Any, List
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_data_science_utils_python.preprocessing.sql.schema.column import \
+    Column
+from exasol_data_science_utils_python.preprocessing.sql.schema.column_name import \
+    ColumnName
+from exasol_data_science_utils_python.preprocessing.sql.schema.column_type import \
+    ColumnType
+from exasol_data_science_utils_python.preprocessing.sql.schema.schema_name import \
+    SchemaName
+from exasol_data_science_utils_python.preprocessing.sql.schema.table_name import \
+    TableName
+
+from exasol_advanced_analytics_framework.event_handler.event_handler_base \
+    import EventHandlerBase
+from exasol_advanced_analytics_framework.event_handler.event_handler_context \
+    import EventHandlerContext
+from exasol_advanced_analytics_framework.event_handler.event_handler_result \
+    import EventHandlerReturnQuery, EventHandlerResultBase, \
+    EventHandlerResultFinished
+from exasol_advanced_analytics_framework.event_handler.event_handler_state \
+    import EventHandlerState
+from exasol_advanced_analytics_framework.event_context.udf_event_context \
+    import UDFEventContext
+
 
 class CreateEventHandlerUDF:
+
     def __init__(self, exa):
         self.exa = exa
 
     def run(self, ctx) -> None:
-        ctx.emit("return query")
-        ctx.emit("status")
-        ctx.emit("query 1")
-        ctx.emit("query 2")
+        # get and set method parameters
+        iter_num = ctx[0]  # iter_num
+        bucketfs_connection = ctx[1]  # bucketfs_connection_name
+        event_handler_class = ctx[2]  # event_handler_class_name
+
+        bucketfs_location = BucketFSFactory().create_bucketfs_location(
+            url=bucketfs_connection.address,
+            user=bucketfs_connection.user,
+            pwd=bucketfs_connection.password
+        )
+        latest_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
+
+        # load the latest (create if not) event handler state object
+        latest_state = self._load_latest_state(
+            ctx, iter_num, event_handler_class,
+            latest_bucketfs_path, bucketfs_location)
+        event_handler_context: EventHandlerContext = latest_state.context
+        event_handler: EventHandlerBase = latest_state.event_handler
+        query_columns: List[Column] = latest_state.query_columns
+
+        # call the user code
+        udf_event_context = self._create_udf_event_context(
+            ctx, iter_num, query_columns)
+        result: EventHandlerResultBase = event_handler.handle_event(
+            udf_event_context, event_handler_context)
+
+        # save the current state and remove the previous state
+        iter_num += 1
+        current_bucketfs_path = PurePosixPath(
+            f"{event_handler_class}_{str(iter_num)}.pkl")
```

Comment:
should we maybe use the fully qualified name of the event handler, meaning with the module name

Reply:
in the iterations after the first iteration, we need event_class_name to read previous state. Because we saved states with the event_class_name.  If we also use event_module_name while saving , we have to add event_module_name to UDF query at each iteration. Do we want to do that?